example/lambdas/db-provision-user-database/src/index.ts

```diff
@@ -41,14 +27,10 @@ export const handler = async (event: HandlerEvent): Promise<void> => {
   let knex;
 
   try {
-    const secretsManager = new AWS.SecretsManager();
-    const config = await getConnectionConfig(event.rootLoginSecret);
-    knex = Knex({
-      client: 'pg',
-      connection: config,
-      acquireConnectionTimeout: 120000,
-    });
-    const dbUser = event?.prefix.replace('-', '_');
+    knex = await getConnectionFromEnvironment(
+      { databaseCredentialSecretId: event.rootLoginSecret }
+    );
+    const dbUser = event?.prefix.replace(/-/g, '_');
```

Comment:
I must have missed this before: so the database username is derived from the prefix? And I assume dashes in the username are not allowed? Also, it seems like `event.prefix` and `event.dbPassword` are required, so should we throw an error if they're missing before we go to validate them?

Reply:
Is there some reason we use the prefix instead of an `event.dbPassword` explicitly? I suppose because this is for internal use only and this makes things easier?

Reply:
Yes - I can't think of a good reason to not have this use a programatic name generation, particularly given it's for internal use. 

>  so should we throw an error if they're missing before we go to validate them?

Yes -  relying on the db for that is crummy:    e94aff2df